File: python-ndn/encode-02.py
import time
from ndn.encoding import make_data, MetaInfo, Component
from ndn.security.signer import DigestSha256Signer


def one_run(i: int):
    name = f"/go-ndn/test/{i:010}"
    content = bytes(4000)
    make_data(name, MetaInfo(0, 4000, Component.from_str("10000")), content, DigestSha256Signer())


def main():
    tic = time.perf_counter()
    for i in range(1000000):
        one_run(i)
    toc = time.perf_counter()
    print(f"Total time: {toc - tic:0.4f} seconds")


if __name__ == '__main__':
    main()

File: ndnts-deno/encode-02.ts
import { Name, Data } from "@ndn/packet"
import { Encoder } from "@ndn/tlv"

const oneRun = (i: number) => {
  const nameStr = "/go-ndn/test/" + i.toLocaleString('en-US', { minimumIntegerDigits: 10, useGrouping: false })
  const content = new Uint8Array(4000)
  const data = new Data(
    new Name(nameStr),
    Data.ContentType(0),
    Data.FreshnessPeriod(4000),
    Data.FinalBlock,
    "10000",
    content
  )
  Encoder.encode(data)
}

if (import.meta.main) {
  const tic = performance.now()
  for (let i = 0; i < 1000000; i++) {
    oneRun(i)
  }
  const toc = performance.now()
  console.log(`Total time: ${(toc - tic) / 1000.0} seconds`)
}
